### PR TITLE
Implement HTTP Status Code callback

### PR DESCRIPTION
**Pull request by ozten**

### CODE REVIEW

File: http_client.rc [hunk 1]
```diff
@@ -268,6 +268,14 @@ pub impl<C: Connection, CF: ConnectionFactory<C>> HttpRequest<C, CF> {
 
     fn on_message_complete(&self) -> bool {
         debug!("on_message_complete");
+        let status_code = self.parser.status_code();
+
+        let status = match status_code {
```
Comment (ozten) [inline http_client.rc:273]: There is probably a very simple idiom here, but I'm just learning Rust.

status_code is an `int` and Status is an `Enum`.

Comment (jdm) [inline http_client.rc:273]: Nope, no idiom.

**Comment by metajack** [comment]
r+ pending typo fix.

File: http_client.rc (hunk 1 above)
Comment (ozten) [inline http_client.rc:273]: Cool, thanks for validating this usage.

**Comment by metajack** [comment]
Thanks!

**Pull request merged**